R/FilterState-abstract.R

```diff
@@ -0,0 +1,70 @@
+#' @name FilterState
```

Comment:
If we stick to our current convention of naming the file after the abstract class (e.g. FilteredDataset.R for the FilteredDataset abstract class), shouldn't we name the file for this abstract class "FilterState.R"? It would be consistent with our naming convention.

I also left a comment on the filename for the InteractiveFilterState class.